### ISSUE TITLE
pytest ( versions 7.2+ ) with pytest-cov 4.0.0 fails to write xml report durring GH actions

### ISSUE THREAD

**Issue by reactive-firewall** [reactive-firewall/multicast#49]
**Describe the bug**

**To Reproduce**
1. ensure working version locally
2. run the `make test-pytest` target
3. ensure coverage output file is created
4. compare to GH actions on any os and any python 3.7-3.10 (3.11 is an expected fail)
5. See error

**Expected behavior**
CI should produce the same or similar behavior to running the same steps locally.

**Logs** _(optional)_
CI builds #138-#150

**Screenshots** _(optional)_

 - Python Version [3.7-3.11]
 - Multicast Python Library Version: 1.4.2

**Additional context** _(optional)_
This seems to be related to only to coverage within pytest at the moment.

**Comment by reactive-firewall** [comment]
Work around for now is to use the M.A.T.s test via `make test` and use `coverage` out-of-band to check test coverage.

**Comment by reactive-firewall** [comment]
🤔 Napkin code for a regression test of this:

```bash
#! /bin/bash
# ---> will need header and boilerplate code here <---
make test-pytest ; wait ;
test -r ./test-reports/junit.xml || EXIT_CODE=1 ;
exit $EXIT_CODE ;
```

**Comment by reactive-firewall** [comment]
✅ Issue seems fixed as of `pytest-cov v5.0+`
